homeassistant/components/modem_callerid/config_flow.py

```diff
@@ -0,0 +1,126 @@
+
+    async def async_step_usb_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle USB Discovery confirmation."""
+        if user_input is not None:
+            return self.async_create_entry(
+                title=user_input.get(CONF_NAME, DEFAULT_NAME),
+                data={CONF_DEVICE: self._device},
+            )
+        self._set_confirm_only()
+        return self.async_show_form(step_id="usb_confirm")
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+        ports = await self.hass.async_add_executor_job(serial.tools.list_ports.comports)
+        for port in ports:
+            if self.source == config_entries.SOURCE_IMPORT:
+                if port.device == user_input.get(CONF_DEVICE):  # type: ignore
+                    uid = f"{port.vid}:{port.pid}_{port.serial_number}_{port.manufacturer}_{port.description}"
+                    entry, errors = await self.validate_input(
+                        user_input=user_input, dev_path=port.device, uid=uid
+                    )
+                    if errors is None:
+                        return entry
+            for entry in self._async_current_entries():
+                if entry.data[CONF_DEVICE] == port.device:
+                    ports.remove(port)
+        list_of_ports = [
+            f"{p}, s/n: {p.serial_number or 'n/a'}"
+            + (f" - {p.manufacturer}" if p.manufacturer else "")
+            for p in ports
+        ]
+        if not list_of_ports:
+            return self.async_abort(reason="no_devices_found")
+
+        if user_input is not None:
+            port = ports[list_of_ports.index(str(user_input.get(CONF_DEVICE)))]
+            dev_path = await self.hass.async_add_executor_job(
+                usb.get_serial_by_id, port.device
+            )
+            uid = f"{port.vid}:{port.pid}_{port.serial_number}_{port.manufacturer}_{port.description}"
+            entry, errors = await self.validate_input(dev_path=dev_path, uid=uid)
+            if errors is None:
+                return entry
+        user_input = user_input or {}
+        schema = vol.Schema({vol.Required(CONF_DEVICE): vol.In(list_of_ports)})
+        return self.async_show_form(step_id="user", data_schema=schema, errors=errors)
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+        if self._async_current_entries():
+            _LOGGER.warning(
+                "Loading Modem_callerid via platform setup is deprecated; Please remove it from your configuration"
+            )
+        if CONF_DEVICE not in config:
+            config[CONF_DEVICE] = DEFAULT_PORT
+
+        return await self.async_step_user(config)
+
+    async def validate_input(self, user_input=None, dev_path=None, uid=None):
```

Comment:
```suggestion
    async def validate_input(self, user_input=None, dev_path=None, unique_id=None):
```

`uid` is confusing here as at first glance, I thought it was a unix user id.